src/benchmarks/micro/libraries/System.Formats.Tar/Perf.Tar.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Formats.Tar.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.NoWASM)]
+    public class Perf_TarWriter
+    {
+        private static readonly string _fileName = "file.txt";
+        private static Dictionary<string, string> _ea;
+        private static MemoryStream _memoryStream;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _memoryStream = new MemoryStream();
+        }
+
+        [GlobalCleanup]
+        public void Cleanup() => _memoryStream.Dispose();
+
+        [GlobalSetup(Targets = new[] { nameof(PaxTarEntry_WriteEntry), nameof(PaxTarEntry_WriteEntry_Async) })]
+        public void SetupPax()
+        {
+            _ea = new Dictionary<string, string>
+            {
+                { "uname", "username" },
+                { "gname", "groupname" },
+                { "uid", "483745" },
+                { "gid", "193783" },
+                { "mtime", "1409547224" },
+                { "ctime", "1409547225" },
+                { "atime", "1409547226" },
+                // real world scenario:
+                { "MSWINDOWS.rawsd", "AQAAgBQAAAAkAAAAAAAAAAAAAAABAgAAAAAABSAAAAAhAgAAAQIAAAAAAAUgAAAAIQIAAA==" }
+            };
+        }
+
+        [Benchmark]
+        public void V7TarEntry_WriteEntry()
+        {
+            V7TarEntry entry = new V7TarEntry(TarEntryType.RegularFile, _fileName);
+            using TarWriter writer = CreateWriter();
+            writer.WriteEntry(entry);
+        }
+
+        [Benchmark]
+        public async Task V7TarEntry_WriteEntry_Async()
+        {
+            V7TarEntry entry = new V7TarEntry(TarEntryType.RegularFile, _fileName);
+            await using TarWriter writer = CreateWriter();
+            await writer.WriteEntryAsync(entry);
+        }
```

Comment:
It seems that the V7 benchmarks are failing:

```log
[2023/05/22 12:39:00][INFO] System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
[2023/05/22 12:39:00][INFO]  ---> System.ArgumentException: Entry type 'RegularFile' not supported in format 'V7'. (Parameter 'entryType')
[2023/05/22 12:39:00][INFO]    at System.Formats.Tar.TarHelpers.ThrowIfEntryTypeNotSupported(TarEntryType entryType, TarEntryFormat archiveFormat, String paramName)
[2023/05/22 12:39:00][INFO]    at System.Formats.Tar.V7TarEntry..ctor(TarEntryType entryType, String entryName)
[2023/05/22 12:39:00][INFO]    at System.Formats.Tar.Tests.Perf_TarWriter.V7TarEntry_WriteEntry_Async() in /datadisks/disk1/work/B3E10A32/w/ADFC090A/e/src/benchmarks/micro/libraries/System.Formats.Tar/Perf.Tar.cs:line 57
```

Should we remove them? Or is it a bug that needs to be fixed?

```suggestion
```

Reply:
It's not a bug in tar, it's a misuse of the V7 format here. The regular file format for V7 uses a different enum value than the rest of the formats.

Reply:
Oops. You now know what my personal account is 😄.